mcp_hydrolix/mcp_server.py

```diff
@@ -137,7 +147,37 @@ def execute_query(query: str):
 
 @mcp.tool()
 def run_select_query(query: str):
-    """Run a SELECT query in a Hydrolix database"""
+    """Run a SELECT query in a Hydrolix time-series database using the Clickhouse SQL dialect.
+       Queries run using this tool will timeout after 30 seconds. Queries should include either a LIMIT
+       clause or a filter based on the primary key (which will be a timestamp) as a performance guard to
+       ensure they return in a reasonable amount of time. Queries should select specific fields and avoid
+       the use of SELECT * to avoid performance issues. The performance guard used for the query should be
+       clearly communicated with the user, and the user should be informed that the query may take a long
+       time to run if the performance guard is not used. When choosing a performance guard, the user's
+       preference should be requested and used if available. When using aggregations, the performance guard
+       should take form of a primary key filter, or else the LIMIT should be applied in a subquery before
+       applying the aggregations.
+       
+       Example query. Purpose: get logs from the `application.logs` table. Primary key: `timestamp`. Performance
+       guard: 10 minute recency filter.
+
+       `SELECT message, timestamp FROM application.logs WHERE timestamp > now() - INTERVAL 10 MINUTES` 
+
+       Example query. Purpose: get the median humidity from the `weather.measurements` table. Primary key:
+       `date`. Performance guard: 1000 row limit, applied before aggregation.
+
+        `SELECT median(humidity) FROM (SELECT humidity FROM weather.measurements LIMIT 1000)`
+
+       Example query. Purpose: get the lowest temperature from the `weather.measurements` table over the
+       last 10 years. Primary key: `date`. Performance guard: date range filter.
+
+       `SELECT min(temperature) FROM weather.measurements WHERE date > now() - INTERVAL 10 YEARS`
+
+       Example query. Purpose: get the app name with the most log messages from the `application.logs` table in the
+       window between new year and valentine's day of 2024. Primary key: `timestamp`. Performance guard: date
+       range filter.
+        `SELECT app, count(*) FROM application.logs WHERE timestamp > '2024-01-01' AND timestamp < '2024-02-14' GROUP BY app ORDER BY count(*) DESC LIMIT 1`
+      """
```

Comment:
should we also discourage full text search and also suggest using LIKE for column name etc?